sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -157,6 +149,32 @@ private static List<ByteBuffer> serialize(List<TelemetryItem> telemetryItems) {
         }
     }
 
+    private static String toJson(List<TelemetryItem> telemetryItems) throws IOException {
+        try (StringWriter debug = new StringWriter()) {
+            for (int i = 0; i < telemetryItems.size(); i++) {
+                JsonWriter jsonWriter = JsonProviders.createWriter(debug);
+                TelemetryItem telemetryItem = telemetryItems.get(i);
+                telemetryItem.toJson(jsonWriter);
+                jsonWriter.flush();
+                if (i < telemetryItems.size() - 1) {
+                    debug.write('\n');
+                }
+            }
+            return debug.toString();
+        }
+    }
+
+    private static String maskIKeys(List<TelemetryItem> telemetryItems, String json) {
+        Set<String> iKeys
+            = telemetryItems.stream().map(TelemetryItem::getInstrumentationKey).collect(Collectors.toSet());
+        for (String instrumentationKey : iKeys) {
+            String maskedIKey
+                = "*" + instrumentationKey.substring(instrumentationKey.length() - 13, instrumentationKey.length() - 1);
```

Comment:
why not include the last character?
```suggestion
                = "*" + instrumentationKey.substring(instrumentationKey.length() - 13);
```